### PR TITLE
Update scan stats with new API request class

### PR DESCRIPTION
**Pull request by jaimejiyepark**

### CODE REVIEW

**Review by awushensky — APPROVED** [review]

File: CardScan/CardScan/Classes/ScanStats.swift [hunk 1]
```diff
@@ -63,6 +63,19 @@ public struct ScanStats {
...
+    func createPayload() -> ScanStatisticsPayload {
+        let scanStats = ScanStatistics(scans: self.scans,
+                                       cardsDetected: self.cardsDetected,
+                                       torchOn: self.torchOn,
+                                       orientation: self.orientation,
+                                       success: self.success,
+                                       duration: self.duration(),
+                                       model: self.model,
+                                       permissionGranted: self.permissionGranted,
+                                       userCanceled: self.userCanceled)
```
Comment (awushensky) [inline CardScan/CardScan/Classes/ScanStats.swift:75]: maybe this is iOS style, but I really don't like hanging indents like this. It's not guaranteed to be a multiple of four and can break easily by changing anything in the first line. instead, I greatly prefer the style:
```
let scanStats = ScanStatistics(
    scans: self.scans,
    ...
    userCanceled: self.userCanceled,
)
```

Comment (jaimejiyepark) [reply CardScan/CardScan/Classes/ScanStats.swift:75]: I think it depends on the style guide of each place. I just looked different companies guides on multi-line styling and a lot of them just use the xcode indentation. But it's not that big of a deal right now so I can change it to how you suggest it :-)

File: CardScan/CardScan/Classes/ScanBaseViewController.swift [hunk 2]
```diff
@@ -402,7 +408,13 @@ public protocol TestingImageDataSource: AnyObject {
         if (self.sendScanStats) {
             // fire and forget
             let scanStatsPayload = self.ocrMainLoop()?.scanStats.createPayload() ?? ScanStats().createPayload()
-            ScanApi.uploadScanStats(payload: scanStatsPayload, completion: { _, _ in })
+            ScanApi.uploadScanStats(payload: scanStatsPayload, completion: { response, error in
+                guard let status = response?.status, status == "ok" else {
```
Comment (awushensky) [inline CardScan/CardScan/Classes/ScanBaseViewController.swift:412]: is this guaranteed to always be lowercase?

**Review by awushensky — APPROVED** [review]

**Comment by jaimejiyepark** [comment]
The integration test relies on this line https://github.com/getbouncer/cardscan-ios/blob/f37f24b041906560242eeb466911e29f5d8da20f/CardScan/CardScan/Classes/API.swift#L78. 

I don't want to change the integration tests right now and in this PR so I added the `lastScanStatsSuccess` value in the codebase but I plan to remove it and update the tests

File: CardScan/CardScan/Classes/ScanBaseViewController.swift (hunk 2 above)
Comment (jaimejiyepark) [reply CardScan/CardScan/Classes/ScanBaseViewController.swift:412]: yea it is https://github.com/getbouncer/cardscan-ios/blob/f37f24b041906560242eeb466911e29f5d8da20f/CardScan/CardScan/Classes/API.swift#L77-L78

**Pull request merged**